extends StaticBody2D

var wall := 0
var model_name := ""




@onready var area: Area2D = $Area2D

func initialize(spawn_position: Vector2, model: String):
	model_name = model
	wall = 0
	global_position = spawn_position
	#print("üü¢ –°—Ç–≤–æ—Ä–µ–Ω–æ [", model_name, "] –Ω–∞ –ø–æ–∑–∏—Ü—ñ—ó: ", global_position)
	G.should_follow_prediction = true
	
	
func _on_area_body_entered(body: Node) -> void:
	if  wall == 0:
		wall = 1
		#print("üõë –ú–æ–¥–µ–ª—å [", model_name, "] –∑–∞—Ñ—ñ–∫—Å—É–≤–∞–ª–∞ [", body, "] –Ω–∞ ", global_position)
		send_collision_data()

func send_collision_data():
	var collision_data = {
		"model": model_name,
		"wall": wall,
		"position": {
			"x": global_position.x,
			"y": global_position.y
		}
	}
	if G:
		G.register_collision(collision_data)


func _on_area_2d_body_entered(body):

	wall = 0
	#print("üü¢ –ú–æ–¥–µ–ª—å [", model_name, "] –Ω–µ –∑–∞—Ñ—ñ–∫—Å—É–≤–∞–ª–∞ —Å—Ç—ñ–Ω–∏ –Ω–∞ ", global_position)
	send_collision_data()



